src/openrct2-ui/windows/SavePrompt.cpp

```diff
@@ -197,46 +220,23 @@ rct_window* WindowSavePromptOpen()
         window_close(*window);
     }
 
+    if (EnumValue(prompt_mode) >= std::size(window_save_prompt_labels))
+    {
+        log_warning("Invalid save prompt mode %u", prompt_mode);
+        return nullptr;
+    }
+
     if (gScreenFlags & (SCREEN_FLAGS_TRACK_DESIGNER | SCREEN_FLAGS_TRACK_MANAGER))
     {
-        widgets = window_quit_prompt_widgets;
         width = WW_QUIT;
         height = WH_QUIT;
     }
     else
     {
-        widgets = window_save_prompt_widgets;
         width = WW_SAVE;
         height = WH_SAVE;
     }
 
-    if (EnumValue(prompt_mode) >= std::size(window_save_prompt_labels))
-    {
-        log_warning("Invalid save prompt mode %u", prompt_mode);
-        return nullptr;
-    }
-    window = WindowCreate<SavePromptWindow>(
+    return WindowCreate<SavePromptWindow>(
         WindowClass::SavePrompt, width, height, WF_TRANSPARENT | WF_STICK_TO_FRONT | WF_CENTRE_SCREEN);
```

Comment:
Instead, do:

```c++
    auto savePromptWindow = std::make_unique<SavePromptWindow>(prompt_mode);
    return WindowCreate<SavePromptWindow>(std::move(savePromptWindow ),
        WindowClass::SavePrompt, width, height, WF_TRANSPARENT | WF_STICK_TO_FRONT | WF_CENTRE_SCREEN);
```

And implement a constructor for `SavePromptWindow` that initializes a private member variable to the prompt_mode
```c++
SavePromptWindow(PromptMode promptMode) : _promptMode(promptMode) {}
```

This will avoid accessing the global on the `OnOpen` method like I mentioned above, while also avoiding repeated logic for setting the prompt_mode.

Reply:
I had to use one of the older-style constructors that returned `rct_window*` to get the code to compile.
Let me know if there's a better approach I can take.